packages/gatsby/src/utils/webpack.config.js

```diff
@@ -408,6 +408,10 @@ module.exports = async (
       )
     }
 
+    if (process.env.gatsby_executing_command === `build` && program.profile) {
```

Comment:
I'm not certain this is the correct way of determining what command was run.

Reply:
I think the preferred approach would be to compare `stage === 'build-javascript'`

Reply:
If we want to target `gatsby build` specifically  we can use ``stage === `build-javascript` && program.profile``.
If we want to target browser bundles for both build and develop we should use ``target === `web` && program.profile``

I'm not exactly how reliable profiling for development bundles (in `gatsby develop`) would be, but it should at least point to potential problems in components while developers are working on their code allowing for faster feedback loop than running `gatsby build --profile` testing, making code changes and rebuilding.

Ultimately developers should validate with `gatsby build`, but for faster iteration it would will be much nicer in `gatsby develop`

Thoughts?

Reply:
Profiling is supported out the box with `gatsby develop` hence I didn't need to do anything. But as you highlighted the validity of the profiling is questionable when it "isn't what users are running".

So I think I've captured the best of both worlds here?

Reply:
Oh, today I learned ;)